deploy/cdk_exec_policy/cdkExecPolicy.yaml

```diff
@@ -253,13 +261,16 @@ Resources:
               - 'codepipeline:GetPipeline*'
               - 'codepipeline:List*'
               - 'codepipeline:DeletePipeline'
-              - 'codestar-notifications:List*'
-              - 'codestar-notifications:DescribeNotificationRule'
               - 'codecommit:Create*'
               - 'codecommit:DeleteRepository'
               - 'codecommit:*Tag*'
               - 'codebuild:*Project*'
-            Resource: '*'
+            Resource: 
+               - !Sub 'arn:${AWS::Partition}:codecommit:${AWS::Region}:${AWS::AccountId}:${EnvironmentResourcePrefix}*'
+               - !Sub 'arn:${AWS::Partition}:codepipeline:${AWS::Region}:${AWS::AccountId}:${EnvironmentResourcePrefix}*'
+               - !Sub 'arn:${AWS::Partition}:codebuild:${AWS::Region}:${AWS::AccountId}:project/${EnvironmentResourcePrefix}*'
+               - !Sub 'arn:${AWS::Partition}:codebuild:${AWS::Region}:${AWS::AccountId}:project/DDKCodePipelineBuildSynthCd*'
```

Comment:
can we just restrict this by `DDK*` prefix

Also what version of data.all are you testing CDK pipeline creation on?